### ISSUE TITLE
SSL issue with cosigner pool?

### ISSUE THREAD

**Issue by joshfraser** [spesmilo/electrum#5181]
I have a 2of3 p2sh multisig with the Cosigner Pool plugin enabled. When I sign a transaction and click "Send to cosigner" I get the following error message (and my co-signers are seeing the same thing on their machines).

```
Failed to send transaction to cosigning pool:
[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:777)
```

I'm using v3.3.4 on MacOS 10.14.3.

**Comment by bauerj** [comment]
Hey,

the cosigner pool is available on https://cosigner.electrum.org/. Do you see any SSL/TLS warnings when opening that website from your Mac?

**Comment by joshfraser** [comment]
I get the following message, but the SSL certificate from Let's Encrypt seems to be fine.

```
Error response
Error code: 501

Message: Unsupported method ('GET').

Error code explanation: HTTPStatus.NOT_IMPLEMENTED - Server does not support this operation.
```

**Comment by bauerj** [comment]
Thanks, that looks good! If you can still reproduce the issue in Electrum I assume this is some general bug unrelated to the cosigner pool.

**Comment by joshfraser** [comment]
Just confirmed on my cosigners machine that we're still seeing the SSL error with the wallet.

Not sure if this would be relevant, but one detail I forgot to mention is that we're all using hardware wallets (mix of Ledgers & Trezors).

**Comment by bauerj** [comment]
Are the other cosigners using macOS, too?

**Comment by joshfraser** [comment]
Yes.

**Comment by SomberNight** [comment]
I can reproduce the issue on MacOS 10.13 with the release binary (3.3.4)
(also tried on Ubuntu 18.04 and Windows 10, works there)

note: the labels plugin is not affected

@joshfraser thank you for reporting the issue; unfortunately even when this gets fixed, it won't work in your case as the cosigner plugin does not support hardware wallets (the decryption of the received message needs access to an EC private key). You can try transmitting files or raw hex transaction strings or QR code images.

**Comment by passio777x** [comment]
Hi, same problem with Mac OS X and Safe-T.
I created 2/2 cosign wallet. 1 with seed/Master-Public-Key and 1 with Safe-T.

I can't sign the transaction with my Safe-T (always unsigned status, even if I have to confirm on the safe T device and type my digit password.)

I can only sign with the electrum wallet, (got 1/2 sign)...

I still unable to send to cosigner once I click on send to cosign, I got message error :

Failed to send transaction to cosigning pool:
[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:777)



I can't send, and I can't export the transaction to sign it manually.


Maybe this is P2SH / P2WPKH/ ... that affect the process...  But I absolutely know nothing about it...
